### ISSUE TITLE
Ivy doesn't respect suggestion order in magit

### ISSUE THREAD

**Issue by itainoam** [hlissner/doom-emacs#1131]
### Observed behavior

Ivy doesn't respect suggestion order in magit. In helm it works.

### Expected behavior
To view branches by order changed you can set this variable:
(setq magit-list-refs-sortby "-creatordate")
`b` `c`  instead list branches alphabetically
In helm it works.

### Steps to reproduce

1.(setq magit-list-refs-sortby "-creatordate")
2.`b` `c`  in magit buffer to list branches by recency

**Comment by hlissner** [comment]
The fix for this is to add an entry in `ivy-sort-functions-alist`, but I don't know which command to use.

Give one or both of these a try:

```elisp
(add-to-list 'ivy-sort-functions-alist '(magit-completing-read))
(add-to-list 'ivy-sort-functions-alist '(magit-read-starting-point))
```

**Comment by itainoam** [comment]
Thank you Henrik.
On first try it didn't work, but I'll try again. There's also `magit-builtin-completing-read`.

**Comment by hlissner** [comment]
You're correct. It doesn't seem we need to change `magit-completing-read-function` anymore, so I've left it to its default as of fa5bebc, which should resolve your issue. Please update and let me know.

**Comment by itainoam** [comment]
Resolved. Thanks!
